beep/protocol/maccor_to_biologic_mb.py

```diff
@@ -1271,3 +1645,61 @@ def set_global_fields(seqs):
     with open(fp, "wb") as f:
         f.write(file_str.encode("ISO-8859-1"))
         print("created {}".format(fp))
+
+
+def add_cycle_nums_to_csvs(csv_and_transition_rule_file_paths):
+    serializer = CycleTransitionDataSerializer()
+    cycle_num = 1
+    for csv_fp, transition_json_fp, csv_outpath in csv_and_transition_rule_file_paths:
+        with open(transition_json_fp, "r") as f:
+            data = f.read()
+            cycle_transition_rules = serializer.parse_json(data)
+
+        df = pd.read_csv(csv_fp, sep=";")
+
+        cycle_num += cycle_transition_rules.adv_cycle_on_start
+
+        prev_seq_num = int(df.iloc[0]["Ns"])
+        prev_loop_num = int(df.iloc[0]["Loop"])
+        cycle_nums = []
+        for _, row in df.iterrows():
```

Comment:
Any benchmark on how long this takes to run?

Reply:
System Specs:
Macbook Pro
16 GB 2400 MHz DDR4
2.6 GHz 6-Core Intel Core i7
SSD Hard Drive

Processing a 1.7gb file takes 10 minutes.

Pretty atrocious in absolute terms. I haven't done much CSV processing in Python so I'm not entirely sure what standard performance for the task is but I imagine once we verify the logic is correct we can speed it up by an order of magnitude.